compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/SingleFileProject.java

```diff
@@ -34,6 +36,8 @@
  */
 public class SingleFileProject extends Project {
 
+    Path targetDir;
```

Comment:
This can be private final

Reply:
I changed it to private Path targetDir. final might not work since the initialization is done inside a try block and we are ignoring the exception entirely. WDYT?